backend/instancemgmt/intance_manager_ttl.go

```diff
@@ -0,0 +1,115 @@
+package instancemgmt
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	gocache "github.com/patrickmn/go-cache"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+const (
+	defaultInstanceTTL             = 24 * time.Hour
+	defaultInstanceCleanupInterval = 48 * time.Hour
+)
+
+// NewTTLInstanceManager creates a new instance manager with TTL-based caching.
+// Instances will be automatically evicted from the cache after the specified TTL.
+func NewTTLInstanceManager(provider InstanceProvider) InstanceManager {
+	return newTTLInstanceManager(provider, defaultInstanceTTL, defaultInstanceCleanupInterval)
+}
+
+func newTTLInstanceManager(provider InstanceProvider, instanceTTL, instanceCleanupInterval time.Duration) InstanceManager {
+	if provider == nil {
+		panic("provider cannot be nil")
+	}
+
+	// Use go-cache for TTL-based caching
+	cache := gocache.New(instanceTTL, instanceCleanupInterval)
+
+	// Set up the OnEvicted callback to dispose instances
+	cache.OnEvicted(func(_ string, value interface{}) {
+		ci := value.(CachedInstance)
+		if disposer, valid := ci.instance.(InstanceDisposer); valid {
+			time.AfterFunc(disposeTTL, func() {
```

Comment:
@iwysiu IIUC we may no longer need to make use of the disposeTTL here - can you confirm if my understanding is correct?

When the instanceTTL expires, the instance will be inaccessible via `Get()` but still exist in memory. When the cleanup interval comes around, as long as that happens a reasonable amount of time after the instanceTTL, then we don't need to worry about anyone have a hold of the instance so we can dispose right away. Also once the instanceTTL expires, there's no way to revive it.

Reply:
I think we should be ok to remove it, but do we think there's a possibility of the following/do we care if it happens?
1) instance fetched
2) instance expired
3) instance cleared (theoretically possible if it just happens at a regular pace?)
4) instance used
I think that's the only possible issue